legion/engine/core/scenemanagement/scenemanager.cpp

```diff
@@ -1,69 +1,124 @@
+        {
+            sceneEntity = sceneEntities[0];
+        }
+        return sceneEntity;
+    }
 
-
-    bool SceneManager::createScene(const std::string& name)
+    bool SceneManager::create_scene(const std::string& name)
     {
-        scene s;
-        s.id = nameHash(name);
-        sceneNames.emplace(s.id, name);
-        auto sceneHandle = m_ecs->createEntity().add_component<scenemanagement::scene>(s);
-        sceneList.emplace(nameHash(name), sceneHandle);
-        return true;
+        ecs::entity_handle sceneEntity = create_scene_entity();
+
+        if (!get_scene(name))
+        {
+            scene s;
+            s.id = nameHash(name);
+            sceneNames.emplace(s.id, name);
+            sceneList.emplace(nameHash(name), sceneEntity);
+        }
+        return SceneManager::save_scene(name, sceneEntity);
     }
 
-    bool SceneManager::createScene(const std::string& name, ecs::entity_handle& ent)
+    bool SceneManager::create_scene(const std::string& name, ecs::entity_handle& ent)
     {
-        if (!ent.has_component<scenemanagement::scene>())
+        if (!ent.has_component<scene>())
         {
             scene s;
             s.id = nameHash(name);
             sceneNames.emplace(s.id, name);
-            auto sceneHandle = ent.add_component<scenemanagement::scene>(s);
+            auto sceneHandle = ent.add_component<scene>(s);
             sceneList.emplace(nameHash(name), sceneHandle);
-
-            std::ofstream outFile("assets/scenes/"+name+".cornflake");
-            serialization::SerializationUtil::JSONSerialize<ecs::entity_handle>(outFile, ent);
+            sceneCount++;
             //true if entity does not have the scene component
-            return true;
+            return save_scene(name, ent);
         }
         //false if it doesn't
         return false;
     }
 
-    bool SceneManager::loadScene(const std::string& name)
+    bool SceneManager::load_scene(const std::string& name)
     {
-        if (SceneManager::getScene(name))
-        {
-            std::ifstream inFile("assets/scenes/" + name + ".cornflake");
-            serialization::SerializationUtil::JSONDeserialize<ecs::entity_handle>(inFile);
-        }
-        else
+        std::string filename = name;
+        if (!common::ends_with(filename, ".cornflake")) filename += ".cornflake";
+
+        std::ifstream inFile("assets/scenes/" + filename);
+
+        doNotCreateEntities = true;
+        auto hry = world.read_component<hierarchy>();
+        log::debug("Child Count Before: {}", hry.children.size());
+        for(auto child : hry.children)
         {
-            std::ifstream inFile("assets/scenes/" + name + ".cornflake");
-            serialization::SerializationUtil::JSONDeserialize<ecs::entity_handle>(inFile);
-            log::warn("Scene " + name + ".cornflake does not exist in our scenelist, but a file does");
+            log::debug("children remaining {}", world.child_count());
+            child.destroy(true);
```

Comment:
honestly, we actually don't want to destroy all world children but only the ones of the currently active main scene if specified that we want to unload the previous scene. If we want to parallel load multiple scenes than we don't want to unload any scenes.

Reply:
once we implement multiple scenes then yes.